test/Rules.js

```diff
@@ -1,252 +1,336 @@
 var assert = require("assert");
 var Rules = require("../lib/Rules");
 
+describe("Rules", function () {
```

Comment:
Sorry if it all seems nit-picky.  The original creation/use of this was for an isomorphic project where performance is really critical.